packages/client/hooks/useOverflowAvatars.ts

```diff
@@ -40,7 +42,7 @@ const useOverflowAvatars = <T extends {id: string}>(
     displayIdx
   })) as ((T & {key: string; displayIdx: number}) | OverflowAvatar)[]
 
-  if (overflowCount > 0) {
+  if (overflowCount > 0 && !isInit) {
```

Comment:
This fixes a bug where the meeting card includes the `AvatarList` space even though there aren't any connected users. To reproduce:
- add a user to a meeting
- see the user in the `AvatarList` in the meeting card
- refresh the page
- remove user from the meeting
- see that the user avatar has not been removed from the meeting card. This is because the `transitionChildren` includes an overflow avatar

Reply:
when you find edge cases like these it makes my heart swell with pride. seriously, great find!

i want you to dig a little deeper into the root cause of this
why is overflow avatar showing up at all? why is maxAvatars incorrect & how can we fix it?
when an overflow avatar is no longer needed, how can we guarantee that it cleans itself up instead of staying stuck on `EXITING`?

Reply:
Thanks! 😄 

In `useTransition`, we store the previous items in a ref. This way, when an item is removed from an array, we can keep track of which item has been removed so that we can add animations.

In `useOverflowAvatars`, `totalItems` is calculated on the initial render whereas `maxAvatars` is counted on the second render. If there is a connected user and we refresh the page, `totalItems` is 1 but `maxAvatars` is the default 0 so an overflow avatar is added. On the second render, `maxAvatars` is updated, `overflowCount` is 0, the `overflowAvatar` is removed from `visibleAvatars` and `useTransition` updates the removed `overflowAvatar`'s status to Exiting. It still exists in `transitioningChildren` meaning the `minHeight` of the card is extended but we can't see the overflow avatar as its status is Exiting.

By cleaning up exiting items in `useTransition` when the component unmounts, the `overflowAvatar` is removed on the second render. Also, by calculating the `minHeight` of the meeting card using transitioning children that are not exiting, I can remove the `exitSpeed` prop! Also, I've added an initial render check to `overflowCount` to make sure this calculation is accurate.